Tests/AsyncHTTPClientTests/HTTPClientTests.swift

```diff
@@ -708,6 +709,23 @@ class HTTPClientTests: XCTestCase {
             }
         }
     }
+    
+    func testProxySOCKS() throws {
```

Comment:
Can we have some more tests here? Can we validate that attempting to speak SOCKS to a server that doesn't speak SOCKS fails? Can we validate that we handle various handshake failure modes appropriately?